addons/arrays/fnc_prettyPrint.sqf

```diff
@@ -0,0 +1,75 @@
+#include "script_component.hpp"
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_prettyPrint
+
+Description:
+    Makes an array easier to read.
+
+Parameters:
+    _array - The array to format <ARRAY>
+    _tab   - The tab char to use (optional, default: tab) <STRING>
+    _depth - Starting indent (optional, default: 0) <NUMBER>
+
+Returns:
+    Formatted string <STRING>
+
+Examples:
+    (begin example)
+        _return = [ [0, 1, ["22", 33, []], 4] ] call CBA_fnc_prettyPrint;
+        // _return ==>
+        // "[
+        //      0,
+        //      1,
+        //      [
+        //          ""22"",
+        //          33,
+        //          []
+        //      ],
+        //      4
+        //  ]"
+        _return = [ [0, 1, ["22", 33, []], 4], ">---" ] call CBA_fnc_prettyPrint;
+        // _return ==>
+        // [
+        // >---0,
+        // >---1,
+        // >---[
+        // >--->---""22"",
+        // >--->---33,
+        // >--->---[]
+        // >---],
+        // >---4
+        // ]
+    (end)
+
+Author:
+   Terra
+---------------------------------------------------------------------------- */
+SCRIPT(prettyPrint);
+params [["_array", [], [[]]], ["_tab", toString[9], [""]], ["_depth", 0, [0]]];
+if (count _array == 0) exitWith {
+    "[]"
+};
+private _lines = ["["];
+
+{
+    private _line = "";
+    for "_i" from 1 to (_depth + 1) do {
+        _line = _tab + _line;
+    };
+    _line = if (_x isEqualType []) then {
+        _line + ([_x, _tab, _depth + 1] call CBA_fnc_prettyPrint);
+    } else {
+        _line + str _x;
+    };
+    if (_forEachIndex != count _array - 1) then {
```

Comment:
`count _array` should be moved out of loop to variable to not execute it on each iteration. Maybe even `count _array - 1` but I'm not sure if it's worth it (the same is for `_depth + 1` above).

Reply:
I agree with `count _array - 1`, this can saved as a variable. Moving `_depth + 1` outside of the loop would add the need for more expressions since the variable is used before and after the loop. The performance gain would depend on the number of subarrays.